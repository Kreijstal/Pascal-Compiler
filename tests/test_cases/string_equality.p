program string_equality;

var
  Input: string;
  Position: Integer;
  Slice: string;

begin
  Input := 'test';
  Position := 1;
  Slice := Copy(Input, Position, 4);

  if Slice = 'true' then
    WriteLn('Found true')
  else
    WriteLn('Not true');

  if Input[Position] <> '"' then
    WriteLn('Not a quote')
  else
    WriteLn('Is a quote');

  if Slice = Input then
    WriteLn('Slice equals input')
  else
    WriteLn('Slice differs from input');

  WriteLn('Done');
end.
