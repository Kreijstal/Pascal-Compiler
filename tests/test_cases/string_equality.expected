Not true
Not a quote
Slice equals input
Done
